### PR TITLE
parseUri support tel uri

### PR DESCRIPTION
**Pull request by xquanluu**
https://github.com/jambonz/jambonz-feature-server/issues/473

### CODE REVIEW

File: lib/sip-parser/parser.js [hunk 1]
```diff
@@ -243,6 +243,25 @@ function parseUri(s) {
...
+
+function parseTelUri(s) {
+  if (typeof s === 'object')
+    return s;
+
+  const re = /^(tel):([+]?[0-9-.()]*)(;phone-context=(.*))?$/;
+  const r = re.exec(s);
+
+  if (r) {
+    return {
```
Comment (davehorton) [inline lib/sip-parser/parser.js:260]: In addition to 'context' you should parse and return 'params'.  Please review the ABNF syntax [here](https://datatracker.ietf.org/doc/html/rfc3966#section-3).

**Review by davehorton — CHANGES_REQUESTED** [review]

File: lib/sip-parser/parser.js (hunk 1 above)
Comment (xquanluu) [reply lib/sip-parser/parser.js:267]: Please check again, I hope I understand you correctly

Comment (davehorton) [reply lib/sip-parser/parser.js:267]: No, not quite.  In the spec, "context" is called out as a specific property and should be returned as such (like you were doing before).  But additionally, the spec provides for other, general params, and these (if present) should be returned in the 'params' object

**Review by davehorton — CHANGES_REQUESTED** [review]

File: lib/sip-parser/parser.js (hunk 1 above)
Comment (xquanluu) [reply lib/sip-parser/parser.js:272]: fixed

File: test/utils.js [hunk 2]
```diff
@@ -9,6 +9,10 @@ process.on('unhandledRejection', (reason, p) => {
 test('utils', (t) => {
   const uri = Srf.parseUri('sip:1234@10.101.10.1;transport=udp');
   t.ok(uri.params.transport === 'udp', 'exposes Srf.parseUri');
+  const telUri = Srf.parseUri('<tel:+1-201-555-0123;phone-context=drachtio.org;ext=1>');
```
Comment (davehorton) [inline test/utils.js:12]: add an example from the audio codes client

**Review by davehorton — CHANGES_REQUESTED** [review]

**Review by davehorton — APPROVED** [review]

**Pull request merged**